pkg/cloudprovider/aws/fleet/capacity.go

```diff
@@ -41,7 +42,8 @@ func (c *Capacity) Create(ctx context.Context, constraints *cloudprovider.Constr
 		return nil, fmt.Errorf("computing bin packing, %w", err)
 	}
 
-	launchTemplate, err := c.vpcProvider.GetLaunchTemplate(ctx, c.spec.Cluster)
+	// TOOD stop hardcoding architecture
+	launchTemplate, err := c.launchTemplateProvider.Get(ctx, c.spec.Cluster, cloudprovider.ArchitectureAmd64)
```

Comment:
I have a PR that's touching on this a little bit. Architecture will come through as part of the cloudprovider.Constraints. No changes necessary here yet, just FYI.

Reply:
Isn't the Packer the only thing that knows what architecture is going to be needed to satisfy the Pod specs? (Maybe I should just wait for the PR though :-))